### PR TITLE
DE45955 - center count badge icons and add symmetrical padding

### PR DESCRIPTION
**Pull request by annabelsegalld2l**
Previously, the count badge was growing to the right as the numbers got larger, skewing the badge to make it unsymmetrical. 

Our approach to solve this is to add a best-effort padding to both sides of the badge. For small numbers, the numbers now grow from the right towards the center of the badge. For larger numbers, the count is centered over the icon.

### CODE REVIEW

File: components/card/card-footer-link.js [hunk 1]
```diff
@@ -99,13 +99,6 @@ class CardFooterLink extends RtlMixin(LitElement) {
 			d2l-count-badge-icon {
 				text-align: initial;
 			}
-			::slotted(d2l-tooltip) {
-				left: calc(-50% + 11px) !important;
```
Comment (annabelsegalld2l) [inline components/card/card-footer-link.js:103]: No longer needed as the count badge is now symmetrical.

File: components/count-badge/count-badge-icon.js [hunk 2]
```diff
@@ -79,15 +76,30 @@ class CountBadgeIcon extends CountBadgeMixin(LitElement) {
...
-		const numberPadding = this.size === 'small' ? '0.55rem' : '0.7rem';
-		const numberStyles = {
+		let numberStyles = {
 			border: '2px solid white',
-			position: 'relative',
-			left: this.dir === 'rtl' ? 0 : 'var(--d2l-count-badge-icon-padding)',
-			right: this.dir === 'rtl' ? 'var(--d2l-count-badge-icon-padding)' : 0,
-			top: numberPadding,
-			visibility: this.skeleton ? 'hidden' : undefined
+			position: 'absolute',
```
Comment (annabelsegalld2l) [inline components/count-badge/count-badge-icon.js:81]: The number bubble now has absolute positioning. This allows us to not include the number in the count-badge-icon width. All the badges will have the same dimensions regardless of the bubble.

File: components/count-badge/count-badge-icon.js [hunk 3]
```diff
@@ -79,15 +76,30 @@ class CountBadgeIcon extends CountBadgeMixin(LitElement) {
...
+		// center long number strings to prevent overflow
+		const centerNumber = this.getNumberString().length >= 4;
+
+		if (centerNumber) {
+			const xPadding = 'var(--d2l-count-badge-icon-padding)';
+			numberStyles = {
+				... numberStyles,
+				transform: this.dir === 'rtl'
+					? `translateY(-50%) translateX(calc(0px - ${xPadding}))`
+					: `translateY(-50%) translateX(${xPadding})`
```
Comment (annabelsegalld2l) [inline components/count-badge/count-badge-icon.js:94]: The translateY moves the bubble to overlap over the icon. The X translation is calculated to center the bubble over the icon (line 34).

**Review by dbatiste — APPROVED** [review]
Looks good!

**Pull request merged**